hail/src/main/scala/is/hail/expr/ir/IsScalarConstant.scala

```diff
@@ -5,18 +5,16 @@ import is.hail.expr.types._
 object IsScalarType {
   def apply(t: Type): Boolean = {
     t match {
-      case TInt32(_) | TInt64(_) | TFloat32(_) | TFloat64(_) | TBoolean(_) => true
+      case TInt32(_) | TInt64(_) | TFloat32(_) | TFloat64(_) | TBoolean(_) | TString(_) => true
```

Comment:
fwiw I think we've been using "scalar" within the IR to indicate that it's being passed around as a value on the stack instead of as a pointer into a region of memory. To that end, I'm kind of inclined to think that `Literal` should just handle the `TString` case and we should get rid of `Str`.

Reply:
I'm also fine keeping `Str` and `Literal` separate, but I don't think there's a super huge difference between strings and the things that `Literal` is designed to handle, and I'd rather keep our use of the word "scalar" consistent at least within this file.

Reply:
I'll change this to `IsEmittable` or something. That's the important difference.

Reply:
ok, sounds good.